Error: set(['f']) called inside main is not defined